haddock-api/src/Haddock/Interface/Create.hs

```diff
@@ -727,7 +728,7 @@ availExportItem is_sig modMap thisMod semMod warnings exportedNames
               -- parents is also exported. See note [1].
               | t `notElem` declNames,
                 Just p <- find isExported (parents t $ unL decl) ->
-                do liftErrMsg $ tell [
+                do liftErrMsg $ tell [ dislocatedErrMsg $
```

Comment:
As before, the `"Warning: "` is superfluous. Let's also pass the `moduleLocation` here.